src/runtimes/node/bundlers/nft/index.ts

```diff
@@ -109,6 +118,19 @@ const traceFilesAndTranspile = async function ({
     ignore: getIgnoreFunction(config),
     readFile: async (path: string) => {
       try {
+        if (extname(path) === '.ts') {
```

Comment:
what about circular dependencies? Any fast lookups for already transpiled files?

```
-> main.ts
  -> imports a.ts
    -> imports b.ts
  -> imports b.ts (already transpiled through a.ts)
```

Reply:
NFT is doing this for us using its own cache. I've tested a scenario similar to yours and verified that we only transpile `b.ts` once.